scripts/fedmsg-map.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+""" Utility to scan a fedmsg setup for port availability.
+
+Reports what percentage of fedmsg endpoints are bound and ready.
+"""
+
+import base64
+import collections
+import socket
+import sys
+import time
+
+import fedmsg.config
+config = fedmsg.config.load_config()
+
+timeout = 0.2
+expected = '/wAAAAAAAAABfw=='
+
+active = collections.defaultdict(list)
+inactive = collections.defaultdict(list)
+
+for_collectd = 'collectd' in sys.argv
+
+
+def info(content="\n"):
+    if not for_collectd:
+        sys.stdout.write(content)
+        sys.stdout.flush()
+
+
+def do_scan():
+    for i, item in enumerate(config['endpoints'].items()):
+        name, endpoints = item
+        for endpoint in endpoints:
+            if not endpoint.startswith('tcp://'):
+                raise ValueError("Don't know how to deal with %r" % endpoint)
+            endpoint = endpoint[len('tcp://'):].split(':')
+            connection = None
+            try:
+                connection = socket.create_connection(endpoint, timeout)
+                actual = base64.b64encode(connection.recv(10))
+                if actual != expected:
+                    inactive[name].append((
+                        endpoint, "%r is not %r" % (actual, expected)))
+                    info("F")
+                else:
+                    active[name].append((endpoint, "all active"))
```

Comment:
I understand the ``active`` but not why it says ``*all* active``, does it loop through the port or so?